scripts/update_changelog_and_credits.py

```diff
@@ -0,0 +1,452 @@
+
+        for char in developer_name_dict:
+            # This case is only for developer names starting with Q since
+            # as of now we have no developers listed whose names start with
+            # a Q.
+            if '%s<span>%s</span>\n' % (
+                    span_indent, char) not in about_page_lines:
+                prev_char = chr(ord(char) - 1)
+                prev_start_index = about_page_lines.index(
+                    '%s<span>%s</span>\n' % (span_indent, prev_char)) + 2
+                prev_end_index = (
+                    prev_start_index + about_page_lines[
+                        prev_start_index:].index('%s</ul>\n' % span_indent))
+                developer_names = sorted(
+                    developer_name_dict[char], key=lambda s: s.lower())
+                span_elem = '%s<span>%s</span>\n' % (span_indent, char)
+                ul_start_elem = '%s<ul>\n' % span_indent
+                ul_end_elem = '%s</ul>\n' % span_indent
+                about_page_lines[prev_end_index + 1:prev_end_index + 1] = (
+                    [span_elem, ul_start_elem] + developer_names + [
+                        ul_end_elem])
+                continue
+
+            start_index = about_page_lines.index(
+                '%s<span>%s</span>\n' % (span_indent, char)) + 2
+            end_index = start_index + about_page_lines[start_index:].index(
+                '%s</ul>\n' % span_indent)
+
+            old_developer_names = about_page_lines[start_index:end_index]
+            updated_developer_names = (
+                old_developer_names + developer_name_dict[char])
+            updated_developer_names = sorted(
+                updated_developer_names, key=lambda s: s.lower())
+            about_page_lines[start_index:end_index] = updated_developer_names
+
+    with python_utils.open_file(ABOUT_PAGE_FILEPATH, 'w') as about_page_file:
+        for line in about_page_lines:
+            about_page_file.write(line)
+    python_utils.PRINT('Updated about-page file!')
+
+
+def remove_updates_and_delete_branch(repo_fork, target_branch):
+    """Remove changes made to AUTHORS, CHANGELOG, CONTRIBUTORS
+    and about-page and delete the branch created with these changes.
+
+    Args:
+        repo_fork: github.Repository.Repository. The PyGithub object for the
+            forked repo.
+        target_branch: str. The name of the target branch.
+    """
+
+    common.run_cmd(GIT_CMD_CHECKOUT)
+    # The get_git_ref code is wrapped in try except block since the
+    # function raises an exception if the target branch is not found.
+    try:
+        repo_fork.get_git_ref('heads/%s' % target_branch).delete()
+    except github.UnknownObjectException:
+        pass
+    except Exception:
+        raise Exception(
+            'Please ensure that %s branch is deleted before '
+            're-running the script' % target_branch)
+
+
+def create_branch(repo_fork, target_branch, github_username):
```

Comment:
Adding test for this function would be basically adding mocks for the pygithub functions like `get_branch`, `create_git_ref` etc just to ensure every line is covered but we cannot check if the branch is actually created.

Reply:
We have two choices:

(a) We could disable this using the no-coverage check -- see my other comment for details.
(b) We could test this trivially using mocking.

I think (b) is probably better, because if you do (a), other devs will just follow it. What do you think? It has the benefit that, if these functions become more complex in the future, we have tests for them that can subsequently be augmented. For now, you could just check that the correct methods are called using a CallCounter (see test_utils).

Reply:
Yes, I think I agree with you on this. (b) is a better choice here. I will add the tests accordingly.

Reply:
Added tests.